ircmsg/tags_test.go

```diff
@@ -65,50 +74,23 @@ func TestUnescape(t *testing.T) {
 // tag string tests
 type testtags struct {
 	raw  string
-	tags map[string]TagValue
-}
-type testtagswithlen struct {
-	raw    string
-	length int
-	tags   map[string]TagValue
+	tags map[string]string
 }
 
-var tagdecodelentests = []testtagswithlen{
-	{"time=12732;re", 512, *MakeTags("time", "12732", "re", nil)},
-	{"time=12732;re", 12, *MakeTags("time", "12732", "r", nil)},
-	{"", 512, *MakeTags()},
-}
 var tagdecodetests = []testtags{
-	{"", *MakeTags()},
-	{"time=12732;re", *MakeTags("time", "12732", "re", nil)},
+	{"", map[string]string{}},
+	{"time=12732;re", map[string]string{"time": "12732", "re": ""}},
+	{"time=12732;re=;asdf=5678", map[string]string{"time": "12732", "re": "", "asdf": "5678"}},
```

Comment:
Any chance we could add a test for a tag with no name, e.g. `time=12314;=413` and make sure the second tag is correctly discarded silently?